src/accelerate/state.py

```diff
@@ -203,10 +204,13 @@ def __init__(
                 self.local_process_index = int(os.environ.get("LOCAL_RANK", -1))
                 self.device = torch.device("cuda", self.local_process_index)
                 torch.cuda.set_device(self.device)
-                self.mixed_precision = (
-                    parse_choice_from_env("MIXED_PRECISION", "no") if mixed_precision is None else mixed_precision
-                )
-
+                if os.environ.get("USE_FSDP", "false") == "true":
+                    self.distributed_type = DistributedType.FSDP
+                    self.mixed_precision = "no"
```

Comment:
We should throw an error if we have mixed_precision different from "no" instead of silently ignoring what the user said in their config.

Reply:
Done.